packages/legacy/src/queries/secrets.ts

```diff
@@ -13,11 +15,69 @@ export const useSecretQuery = (
   const secretResource = createSecretResource(namespace);
   return useMockableQuery<ISecret>(
     {
-      queryKey: ['secrets', secretName],
+      queryKey: ['secret', secretName],
       queryFn: async () => await consoleFetchJSON(secretResource.namedPath(secretName)),
       refetchInterval: usePollingContext().refetchInterval,
       enabled: !!secretName,
     },
-    secretName ==='secure' ? MOCK_SECRET_SECURE : MOCK_SECRET_INSECURE
+    secretName === 'secure' ? MOCK_SECRET_SECURE : MOCK_SECRET_INSECURE
   );
 };
+
+// function decodeSecretData (secret: ISecret) {
+//   if (secret.data) {
+//     for (const [key, value] of Object.entries(secret.data)) {
+//       secret.data[key] = atob(value);
+//     }
+//   }
+//   return secret;
+// }
+
+export const useSecretsQuery = (
+  secretNames: string[] | null,
+  namespace: string,
+  // decodeData: boolean = false,
+): UseQueryResult<IKubeList<ISecret>> => {
+  const filterList = useCallback(
+    (data: IKubeList<ISecret>): IKubeList<ISecret> => ({
+      ...data,
+      items: data.items.filter((d) => secretNames.includes((d.metadata as IMetaObjectMeta)?.name)),
+    }),
+    [secretNames]
+  );
+
+  // const filterAndDecode = useCallback(
+  //   (data: IKubeList<ISecret>): IKubeList<ISecret> => {
+  //     const filtered = filterList(data);
+  //     filtered.items = filtered.items.map(decodeSecretData)
+  //     return filtered;
+  //   },
+  //   [filterList]
+  // );
+
+  const secretResource = createSecretResource(namespace);
+  return useMockableQuery<IKubeList<ISecret>>(
+    {
+      queryKey: ['secrets', secretNames],
+      queryFn: async () => await consoleFetchJSON(secretResource.listPath()),
+      refetchInterval: usePollingContext().refetchInterval,
+      enabled: !!secretNames && secretNames.length > 0,
+      // select: decodeData ? filterList : filterAndDecode,
+      select: filterList,
+    },
+    mockKubeList([MOCK_SECRET_SECURE, MOCK_SECRET_INSECURE], 'SecretList')
+  );
+};
+
+export const useSecretsForProvidersQuery = (
+  providersQuery: UseQueryResult<IKubeList<IProviderObject>> | null,
```

Comment:
can we pass a non-null array of providers here?

Reply:
Nope.  Good point.